tests/functional/flakes/common.sh

```diff
@@ -2,6 +2,8 @@
 
 source ../common.sh
 
+export _NIX_TEST_BARF_ON_UNCACHEABLE=1
```

Comment:
I can't help but wonder if we want to invert this -- make it barf on uncacheable when we actually care about that being true...? Mostly just seeing all the places you needed to unset this made me wonder if there would be less "noise" if we did it the other way.

Reply:
No if we did it the other way around, there would be way more places that would need to be marked. And we would be likely to miss a bunch.